### PR TITLE
Support Kubevirt v0.36+

### PR DESCRIPTION
**Pull request by shengnuo**
This PR makes kubevirt-gpu-device-plugin compatible with Kubevirt v0.36 and above.
Going forward, we are dropping the support for legacy Kubevirt (Kubevirt v0.35 or below). Those who wishes to maintain compatibility with older Kubevirt should use v1.0.1.

Fixes https://github.com/NVIDIA/kubevirt-gpu-device-plugin/issues/19

### CODE REVIEW

File: pkg/device_plugin/generic_device_plugin.go [hunk 1]
```diff
@@ -48,6 +48,9 @@ const (
 	DeviceNamespace   = "nvidia.com"
 	connectionTimeout = 5 * time.Second
 	vfioDevicePath    = "/dev/vfio"
+	gpuType           = "VFIO-PCI"
```
Comment (sseetharaman6) [inline pkg/device_plugin/generic_device_plugin.go:51]: nit: not used anywhere?

File: README.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ This is a kubernetes device plugin that can discover and expose GPUs and vGPUs o
 ## Prerequisites
 - Need to have Nvidia GPU configured for GPU pass thorugh or vGPU. Quickstart section provides details about this
 - Kubernetes version >= v1.11
-- Kubevirt release >= v0.22.0
+- Kubevirt release >= v0.36.0
 - Kubevirt GPU feature gate should be enabled. Feature gate is enabled by creating a ConfigMap. ConfigMap yaml can be found under /examples
```
Comment (sseetharaman6) [inline README.md:23]: the configmap example needs to be updated. 
also good to call out in the README that if anyone wants to use  Kubevirt versions < v0.35.0 , then they would have to use deprecated image.

**Comment by zvonkok** [comment]
In recent Ubuntu versions VFIO is builtin hence the configuration needs to be done via kernel cmdline, do we want it update the Readme regarding this as well?

File: README.md (hunk 2 above)
Comment (shengnuo) [reply README.md:25]: good point, updated

File: pkg/device_plugin/generic_device_plugin.go (hunk 1 above)
Comment (shengnuo) [reply pkg/device_plugin/generic_device_plugin.go]: removed

**Comment by cdesiniotis** [comment]
+1 to updating README with VFIO configuration details. Also, Kubevirt now requires that you explicitly [whitelist host devices](https://kubevirt.io/user-guide/virtual_machines/host-devices/#listing-permitted-devices) that you want to be exposed in the cluster. Should we add that somewhere in the README?

**Comment by shengnuo** [comment]
> +1 to updating README with VFIO configuration details. Also, Kubevirt now requires that you explicitly [whitelist host devices](https://kubevirt.io/user-guide/virtual_machines/host-devices/#listing-permitted-devices) that you want to be exposed in the cluster. Should we add that somewhere in the README?

Yup, that would be a good idea

**Comment by shengnuo** [comment]
> In recent Ubuntu versions VFIO is builtin hence the configuration needs to be done via kernel cmdline, do we want it update the Readme regarding this as well?

hey @zvonkok, thanks for raising this.
Just to clarify, are we talking about the [VFIO configuration steps](https://github.com/NVIDIA/kubevirt-gpu-device-plugin#2-enable-vfio-pci-kernel-module) no longer applies to Ubuntu 20?

**Comment by rthallisey** [comment]
lgtm

**Comment by zvonkok** [comment]
@shengnuo Yes, the VFIO_PCI is not a module per default in a Ubuntu 20.04
```
CONFIG_KVM_VFIO=y
CONFIG_VFIO_IOMMU_TYPE1=y
CONFIG_VFIO_VIRQFD=y
CONFIG_VFIO=y
CONFIG_VFIO_NOIOMMU=y
CONFIG_VFIO_PCI=y
CONFIG_VFIO_PCI_VGA=y
CONFIG_VFIO_PCI_MMAP=y
CONFIG_VFIO_PCI_INTX=y
CONFIG_VFIO_PCI_IGD=y
CONFIG_VFIO_MDEV=m
CONFIG_VFIO_MDEV_DEVICE=m
```
To configure the GPU for passthrough one needs to update the kernel cmdline ` vfio-pci.ids=10de:1eb8` rather then creating a config for the module.

**Pull request merged**